src/components/SettingsPage/PasswordAutoLockSelect.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import React, { PureComponent } from 'react'
+import { translate } from 'react-i18next'
+import { createStructuredSelector } from 'reselect'
+import { connect } from 'react-redux'
+import { setAutoLockTimeout, saveSettings } from 'actions/settings'
+import Select from 'components/base/Select'
+import type { T } from 'types/common'
+import { autoLockTimeoutSelector } from 'reducers/settings'
+
+type Props = {
+  autoLockTimeout: string,
+  setAutoLockTimeout: (?number) => void,
+  t: T,
+}
+
+const mapStateToProps = createStructuredSelector({
+  autoLockTimeout: autoLockTimeoutSelector,
+})
+
+const mapDispatchToProps = {
+  saveSettings,
+  setAutoLockTimeout,
+}
+
+class PasswordAutoLockSelect extends PureComponent<Props> {
+  handleChangeTimeout = ({ value: timeoutKey }: *) => {
+    this.props.setAutoLockTimeout(+timeoutKey)
+  }
+
+  timeouts = [
+    { value: '1', label: `1 ${this.props.t('app:time.minute')}` },
+    { value: '10', label: `10 ${this.props.t('app:time.minute')}s` },
+    { value: '30', label: `30 ${this.props.t('app:time.minute')}s` },
+    { value: '60', label: `1 ${this.props.t('app:time.hour')}` },
+    { value: '-1', label: this.props.t(`app:common.never`) },
```

Comment:
kinda minor feedback here, but it should be possible to make `value` a `number` , that way you don't need to `+()` cast it. it might just be about providing a render item function for `<Select>`